### ISSUE TITLE
AbstractBundle - getBasePath - missing last letter

### ISSUE THREAD

**Issue by Kaik** [zikula/core#3862]
| Q                 | A
| ----------------- | ---
| Zikula Version    | 1.5.x
| PHP Version       | 7x

### Expected behavior
full base path

### Actual behavior
missing the last letter in the base path

### Steps to reproduce
```
dump($this->kernel->getModule('ZikulaUsersModule')->getBasePath());

"/home/Kaik/public_html/core15/src/system/UsersModul"
```
I do not know if it is on purpose or OS dependent. I just noticed it while working on Dizkus and eventually used getNamespace.

**Comment by Kaik** [comment]
In controller: (core 1.5.2 and 2.0.3)
 ```  
dump($this->get('kernel')->getModule('ZikulaUsersModule')->getBasePath());
result:
"/home/Kaik/public_html/mojeleicester/src/system/UsersModul"

dump($this->get('kernel')->getModule('ZikulaUsersModule')->getNamespace());
result:
"Zikula\UsersModule"

dump($this->get('kernel')->getModule('ZikulaUsersModule')->getPath());
result:
"/home/Kaik/public_html/mojeleicester/src/system/UsersModule"
```
As I wrote before I'm using getNamespace it works and it fits better for what I need, but during testing those methods getBasePath missing last letter brought my attention.

**Comment by Guite** [comment]
@kaik can you please try
replacing https://github.com/zikula/core/blob/1.5/src/lib/Zikula/Core/AbstractBundle.php#L148
by:
```
$this->basePath = substr($path, 0, strrpos($path, $ns));
$this->basePath = rtrim($this->basePath, '/');
```

Does this fix the problem?

**Comment by Kaik** [comment]
@Guite no luck, unfortunately:
Method after changes looks like this hope this is what you meant
```
    public function getBasePath()
    {
        if (null === $this->basePath) {
            $ns = str_replace('\\', '/', $this->getNamespace());
            $path = str_replace('\\', '/', $this->getPath());
            $this->basePath = substr($path, 0, strrpos($path, $ns));
	    $this->basePath = rtrim($this->basePath, '/');
        }

        return $this->basePath;
    }
``` 
And the result is an empty string "". I have tried onliner as well - my ide is complaining when I do assign variable twice but `` $this->basePath = rtrim(substr($path, 0, strrpos($path, $ns)), '/'); `` returns exactly the same result an empty string.

**Comment by Guite** [comment]
How about this:
```
$basePath = substr($path, 0, strrpos($path, $ns));
$this->basePath = substr($basePath, -1) == '/' ? substr($basePath, 0, -1) : $basePath;
```
?

**Comment by Guite** [comment]
Seems like this whole `getBasePath` method is not solid because it assumes the namespace is part of the path.
Honestly I would like to deprecate and remove it (looks like it isn't used anywhere in the core, too).

What exactly do you expect from it? `/home/Kaik/public_html/core15/src/system/`?

@craigh opinion?

**Comment by craigh** [comment]
I agree it is probably no longer useful. It was probably used in PSR-0 days, but now we are requiring PSR-4 in Core-2+ and the namespace is not reliable as a path. Cannot remove until 3.0 though

**Comment by Kaik** [comment]
>What exactly do you expect from it? /home/Kaik/public_html/core15/src/system/?

Well, I needed module root path so whatever is before ModuleName I have used a different method for my purposes. In the meantime, I have noticed that last letter is missing in the output path so I created this issue. 
Deprecating it would be very helpful I do avoid deprecated methods so the code I do is more future proof.

**Comment by craigh** [comment]
_why_ do you need module root path? Frankly, you shouldn't be using it.

**Comment by Kaik** [comment]
I need it for HookedTopicMeta here https://github.com/Kaik/Dizkus/blob/hooks_core2/Hooks/TopicProBundle.php#L504

**Comment by craigh** [comment]
obviously a file path wouldn't work in that situation anyway.

I would recommend you use tagged services for that purpose - not the old method.

**Comment by Kaik** [comment]
Good I will go for tagged services when I will be back working on Dizkus.

**Comment by craigh** [comment]
OK I did some experimenting with this. In Zikula 1.4.6 the result is the same as the OP (missing last letter). Even back in 1.4.6, Users was a psr-4 module. So I searched far and wide for a psr-0 module and only found an old release of Dizkus. Indeed, the method returned an appropriate value for Dizkus. I also tried it with a 1.3-era module - PostCalendar and the method failed completely because old style modules are not registered with the kernel.

So, my solution is to make some notes in the 1.5 release cycle about the issue and then immediately remove it from the 2.0 release since psr-0 modules are not supported there. technically, this is a BC break, but I don't see it as a problem because it would never work anyway and the `->getPath()` method is what should be used.
